### ISSUE TITLE
Activity to login in OKTA is not opening from time to time

### ISSUE THREAD

**Issue by akolubau** [okta/okta-oidc-android#211]
I've faced very strange behavior of OKTA library. Sometimes OKTA login screen is not appearing at all. The same app build on a few different devices behaves differently: on one device works as expected, on another device, web view for login to OKTA is not opening at all. Sometimes I receive ResultCallback.onError with null as AuthorizationException param, sometimes login screen is not opening at all. What is stranger, is that if I clean app data (and leave the same app build) on both devices, behavior could change to vice-versa: the first device (that worked as expected) stopped to work, the second device (where the login screen was not appearing) starts to work normal. 

At the beginning I thought that the problem is in having 2 builds (release and debug) with the same
manifestPlaceholders = [
    "appAuthRedirectScheme": "com.mypackage"
]
But this suggestion was refuted after plenty of tests. 

Any ideas/suggestions what could be wrong and could cause this behavior?

**Comment by JayNewstrom** [comment]
My guess (related to #212) is that you're making calls in parallel, when the SDK was designed to be made in sequence. Or there are threading issues.

If you can give us a reproducible sample (or modify one of our samples to exhibit the same behavior), I'd be more than happy to look into it deeper.

**Comment by akolubau** [comment]
@JayNewstrom , TY for your help. After making sure, that threads work correctly, I was unable to face any problem or unexpected behavior. However, our QA still facing the same problem permanently for all his devices. After hard reset to factory settings one of his device works correctly, but another one even after reset issuing the same problem. 

Just letting you know, that we've got significant progress, and I'll continue to gather information for fix attempt or addressing it to you.

**Comment by JayNewstrom** [comment]
Glad to hear you're making progress! If you've got something reproducible on my end, I'd be happy to take a look.

**Comment by jacobp100** [comment]
Can I ask - has anyone tried experimenting a custom/native login screen? We've seen this resolves the issue, although our business still doesn't want to go that route.

We started looking into the code for opening up a chrome custom tab, although our debugging process for this is very slow.

We've 'fixed' the issue for some users by asking them to use Chrome as the default browser and also restarting their phone. However, this did not work for all users.

**Comment by JayNewstrom** [comment]
We've got an [Android Sample](https://github.com/okta/samples-android/tree/master/sign-in-kotlin) using [okta-auth-java](https://github.com/okta/okta-auth-java) which allows you to build a custom native UI talking directly to our auth APIs.

We've had better luck using `.browserMatchAll(true)` https://github.com/okta/okta-oidc-android/blob/363015cc58c57f7e662f4cf3a63e724665682dea/library/src/main/java/com/okta/oidc/Okta.java#L117 for when users don't have chrome set as their default browser.

**Comment by akolubau** [comment]
@jacobp100  @JayNewstrom is it normal behavior, that Okta's activity stays as a separate task after successful (or failed) login ? Or maybe again I'm missing smth?

**Comment by JayNewstrom** [comment]
You can override the behavior of how the login launches with something similar to what I answered here: https://github.com/okta/okta-oidc-android/issues/225#issuecomment-807231904

**Comment by akolubau** [comment]
TY
